src/do_unmount/do_unmount.c

```diff
@@ -0,0 +1,382 @@
+    RC_STRING *path;              /* Path to unmount */
+    pthread_mutex_t unmounting;   /* Mutex locked until the path is unmounted */
+    global_args_t *global_args;   /* Parameters shared among all threads */
+} thread_args_t;
+
+/* Pass arguments as vector of string to a command and open standard output as readable file */
+static FILE *
+popen_vec(const char *command, int argc, char **argv)
+{
+    FILE *fp;   /* File pointer to program output */
+    char *cmd;  /* Command with all arguments */
+    int length; /* Length of the command with all arguments */
+    int i;      /* Iterator */
+
+    /* Calculate the length of the command */
+    length = strlen(command) + 1; /* +1 for the null terminator */
+    for (i = 0; i < argc; i++)
+        length += strlen(argv[i]) + 3; /* +3 for the quotes and space */
+
+    /* Allocate memory for the command */
+    cmd = xmalloc(length * sizeof(char));
+    sprintf(cmd, "%s", command);
+
+    /* Craft the command with all passed arguments */
+    for (i = 0; i < argc; i++) {
+        strcat(cmd, " \"");
+        strcat(cmd, argv[i]);
+        strcat(cmd, "\"");
+    }
+
+    /* Open the command, free memory and return output file pointer for reading */
+    fp = popen(cmd, "r");
+    free(cmd);
+    return fp;
+}
+
+/* Populate the list of shared mounts in the system */
+static void
+populate_shared_list(RC_STRINGLIST **list)
+{
+    FILE *fp;               /* File pointer to the mountinfo file */
+    size_t len = 0;         /* Length of the line read */
+    char *line = NULL;      /* Line read from the mountinfo file */
+    char *token = NULL;     /* Token of the current line, no need to free this since is pointing to the original string */
+    char *path = NULL;      /* Path to relative to the current line */
+    int i;                  /* Fields iterator */
+
+    /* Initialize list */
+    *list = rc_stringlist_new();
+
+    /* Only linux supports shared mounts, so leave the list empty for eny other OS */
+    #ifdef __linux__
+    /* Open mountinfo file for reading */
+    fp = fopen("/proc/1/mountinfo", "r");
+    if (fp == NULL)
+        exit(EXIT_FAILURE);
+
+    /* Read the output a line at a time */
+    while (getline(&line, &len, fp) != -1) {
+        /* Split line by space */
+        token = strtok(line, " ");
+        i = 0;
+        while (token != NULL) {
+            if (i == 4)
+                xasprintf(&path, "%s", token);  /* Copy token into path */
+            else if (i == 6 && strstr(token, "shared:") != NULL) /* If token contains "shared:" */
+                rc_stringlist_add(*list, path);                  /* Note: if there is no optional field this token will be the separator character '-' */
+            token = strtok(NULL, " ");
+            i++;
+        }
+    }
+
+    /* Close file and free memory */
+    fclose(fp);
+    if (line)
+        free(line);
+    if (path)
+        free(path);
```

Comment:
`free` is no-op on `NULL` argument. No need to check before calling.

Reply:
I didn't know, it's fixed now.